src/hooks/usePollDatasetStatus.js

```diff
@@ -0,0 +1,81 @@
+import { useEffect, useState, useRef } from 'react'
+import { useDatasetManager } from 'hooks/useDatasetManager'
+
+export const usePollDatasetStatus = () => {
+  const {
+    datasetAccessions,
+    setDatasetAccessions,
+    processingDatasets,
+    setProcessingDatasets,
+    getDataset
+  } = useDatasetManager()
+  const [polledDatasetId, setPolledDatasetId] = useState(null)
+  const [polledDatasetState, setPolledDatasetState] = useState(false)
+  const timerRef = useRef(null)
+  const isProcessingDataset = polledDatasetState?.is_processing
+
+  // polls the latest state of the processing dataset per minute
+  // (the processing usually takes a few minutes)
+  useEffect(() => {
+    if (polledDatasetId) {
```

Comment:
should eagerly clear the interval if `polledDatasetId` is true. This would prevent creating a new interval and losing the reference to the old one.

Reply:
See update at 20607db98fa01e5661ccbbc8704d7c1f132774c5.